cookbooks/aws-parallelcluster-common/resources/network_service/partial/_restart_network_debian.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+#
+# Copyright:: 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+
+action :restart_network do
+  network_service_name = 'systemd-resolved'
+  Chef::Log.info("Restarting '#{network_service_name}' service, platform #{node['platform']} '#{node['platform_version']}'")
+
+  service network_service_name.to_s do
+    action %i(restart)
+    ignore_failure true
```

Comment:
Why is it safe to ignore failures here? I know that it's the same code that it was previously in the `libraries/helpers.rb`, but I was wondering what the rationale was.

Reply:
I experienced that if you detach a networking interface and attach a new one you can have issues like:
``` 
Feb 27 11:54:53 ip-10-0-1-234.eu-west-1.compute.internal network[20916]: Bringing up interface eth3:  ERROR     : [/etc/sysconfig/network-scripts/ifup-eth] Device eth3 does not seem to be present, delaying initialization.
```
and in cases like this the restart of networking services exits with error code 1.